web/src/message_events.js

```diff
@@ -177,62 +177,60 @@ export function update_messages(events) {
 
     for (const event of events) {
         const msg = message_store.get(event.message_id);
-        if (msg === undefined) {
```

Comment:
The `msg` variable is reused in a bunch of the code below in a way that's quite terrible; I noticed this while trying to read this. 

Changed it in a new prep commit, since I think the impact of that bug was broadened somewhat.